source/game.betty.bmx

```diff
@@ -317,32 +318,49 @@ Type TBetty
 	End Function
 
 
-	Function CalculateProgrammeScore:int(broadcastMaterial:TBroadcastMaterial)
-		if not broadcastMaterial or broadcastMaterial.owner < 0 then return 0
+	Function CalculateProgrammeScore:int(broadcastMaterial:TBroadcastMaterial, hour:Int = -1)
+		If Not broadcastMaterial Or broadcastMaterial.owner < 0 Then Return 0
 		'not of interest for Betty?
-		if broadcastMaterial.SourceHasBroadcastFlag(TVTBroadcastMaterialSourceFlag.IGNORED_BY_BETTY) then return 0
+		If broadcastMaterial.SourceHasBroadcastFlag(TVTBroadcastMaterialSourceFlag.IGNORED_BY_BETTY) Then Return 0
 
 		'calculate score:
 		'a perfect Betty programme would give 100 points
 		'- topicality<1.0 and rawQuality<1.0 reduce points -> GetQuality()
-		'- "CallIn/Trash/Infomercials" is someting Betty absolutely dislikes
-
-		if TAdvertisement(broadcastMaterial) then return -5
-		local programme:TProgramme = TProgramme(broadcastMaterial)
-		if programme.data.HasFlag(TVTProgrammeDataFlag.PAID) then return -5
-		if programme.data.HasFlag(TVTProgrammeDataFlag.TRASH) then return -3
-		if programme.data.GetGenre() = TVTProgrammeGenre.Erotic then return -20
-		if programme.data.HasSubGenre(TVTProgrammeGenre.Erotic) then return -10
+		'- "CallIn/Trash/Infomercials/Erotic" is someting Betty absolutely dislikes
+		Local points:Float = 0
+		Local pointsMod:Float = 1.0
+		If TAdvertisement(broadcastMaterial)
+			points = -5
+		Else
+			Local programme:TProgramme = TProgramme(broadcastMaterial)
+			Local blocks:Int = programme.GetBlocks()
+			If programme.data.GetGenre() = TVTProgrammeGenre.Erotic 
+				points = -20 * blocks
+			ElseIf programme.data.HasSubGenre(TVTProgrammeGenre.Erotic)
```

Comment:
Und was ist mit Erotik "Zielgruppe Frauen" ? Ignorieren (der Einfachheit halber) ?

Reply:
Das wäre tatsächlich neckisch. Ich muss mal schauen, wieviele Programme/Drehbücher es mit dieser Konstellation gibt.

Ohne diese Anpassung könnte man Ähnliches erreichen, wenn man das bettyIgnore-Flag setzt und einen positiven Betty-Effekt in die Datenbank packt (dann natürlich ohne die Anpassungen für Ausstrahlungszeit...).

Eine Änderung (oder ein Kommentar) nach der Analyse folgt.

Reply:
Tja, aktuell gibt es genau ein Programm (und kein Drehbuch) mit Genre Erotik und Zielgruppe Frauen (in Kurven der Welt). Bei dem ist das ignore-Betty-Flag gesetzt.

Ich denke, es wäre nicht ganz fair (wenn auch lustig), wenn es automatisch Betty-Punkte für Erotikprogramme gäbe, nicht aber für andere Genre, bei denen Frauen als Zielgruppe gesetzt sind. Man könnte natürlich generell Punkte für Zielgruppe Frauen (aber nicht Männer) vergeben, was aber ein ziemlicher Eingriff in die aktuelle Punktevergabe wäre.

Keine weiteren Änderungen und man könnte DB-Effekte definieren?

Reply:
Naja mir ging es nur darum, dass ein Frauenerotikfilm (auch wenn mir da nur 9 q/2 Wochen, 50 Shades usw. in den Sinn kommen) vlt keinen "Abzug" gibt (wenn er Trash ist, dann natuerlich trotzdem..aber nur fuer Trash).


Ansonsten gibt es mein OK.

Reply:
Ist eingebaut.